modules/instance_template/variables.tf

```diff
@@ -365,6 +365,12 @@ variable "enable_confidential_vm" {
   description = "Whether to enable the Confidential VM configuration on the instance. Note that the instance image must support Confidential VMs. See https://cloud.google.com/compute/docs/images"
 }
 
+variable "confidential_instance_type" {
+  type        = string
+  default     = ""
```

Comment:
Will the API accept an empty string?  Might be better to use `null`?

"One of the following values: `SEV`, `SEV_SNP`."